lib/runtime/wasmer/instance.go

```diff
@@ -311,10 +312,10 @@ func (in *LegacyInstance) NetworkService() runtime.BasicNetwork {
 
 // int64ToPointerAndSize converts an int64 into a int32 pointer and a int32 length
 func int64ToPointerAndSize(in int64) (ptr int32, length int32) {
-	return int32(in), int32(in >> 32)
+	return int32(in & 0xffffffff), int32(in >> 32)
```

Comment:
why is this changed?

Reply:
I was experimenting with bitwise operation. Reverted it.